packages/premint-sdk/src/premint-api.ts

```diff
@@ -0,0 +1,684 @@
+    if (response.status !== 200) {
+      let json;
+      try {
+        json = await response.json();
+      } catch (e: any) {}
+      throw new BadResponse(
+        `Invalid response, status ${response.status}`,
+        response.status,
+        json,
+      );
+    }
+    return await response.json();
+  }
+
+  /**
+   * A simple fetch() wrapper for HTTP post.
+   * Can be overridden as needed.
+   *
+   * @param path Path to run HTTP JSON POST against
+   * @param data Data to POST to the server, converted to JSON
+   * @returns JSON object response
+   * @throws Error when HTTP response fails
+   */
+  async post(path: string, data: any) {
+    const response = await fetch(path, {
+      method: "POST",
+      headers: {
+        "content-type": "application/json",
+        accept: "application/json",
+      },
+      body: JSON.stringify(data),
+    });
+    if (response.status !== 200) {
+      let json;
+      try {
+        json = await response.json();
+      } catch (e: any) {}
+      throw new BadResponse(
+        `Bad response: ${response.status}`,
+        response.status,
+        json,
+      );
+    }
+    return await response.json();
+  }
+
+  /**
+   * Getter for public client that instantiates a publicClient as needed
+   *
+   * @param publicClient Optional viem public client
+   * @returns Existing public client or makes a new one for the given chain as needed.
+   */
+  getPublicClient(publicClient?: PublicClient): PublicClient {
+    if (publicClient) {
+      return publicClient;
+    }
+    return createPublicClient({ chain: this.chain, transport: http() });
+  }
+
+  /**
+   * Update premint
+   *
+   * Updates existing premint
+   * @param settings Settings for the new premint
+   * @param settings.account Account to sign the premint update from. Taken from walletClient if none passed in.
+   * @param settings.collection Collection information for the mint
+   * @param settings.walletClient viem wallet client to use to sign
+   * @param settings.uid UID
+   * @param settings.token Mint argument settings, optional settings are overridden with sensible defaults.
+   *
+   */
+  async updatePremint({
+    walletClient,
+    uid,
+    collection,
+    token,
+    account,
+  }: {
+    walletClient: WalletClient;
+    uid: number;
+    token: MintArgumentsSettings;
+    account?: Account | Address;
+    collection: Address;
+  }) {
+    const signatureResponse = (await this.get(
+      `${ZORA_API_BASE}signature/${
+        this.network.zoraBackendChainName
+      }/${collection.toLowerCase()}/${uid}`,
+    )) as PremintResponse;
+
+    const convertedPremint = convertPremint(signatureResponse.premint);
+    const signerData = {
+      ...signatureResponse,
+      premint: {
+        ...convertedPremint,
+        tokenConfig: {
+          ...convertedPremint.tokenConfig,
+          ...token,
+        },
+      },
+    };
+
+    return await this.signAndSubmitPremint({
```

Comment:
minor but you can remove the `await` here since its returns a promise

Reply:
While it's technically redundant does having it here help with readability since someone at a glance will know it's an async function?